Features/Inventory/BigInventory.cs

```diff
@@ -0,0 +1,365 @@
+		{
+			if (index < inventory.Size)
+			{
+				inventory.SetItem(index, item);
+				return;
+			}
+
+			index -= inventory.Size;
+		}
+	}
+
+	public Dictionary<int, ItemStack> AddItem(params ItemStack[] items)
+	{
+		var overflow = new Dictionary<int, ItemStack>();
+		for (var index = 0; index < items.Length; index++)
+		{
+			var item = items[index];
+			var inventory = _inventories[item.Category];
+			var leftover = inventory.AddItem(item);
+
+			// Assertion, should never happen
+			if (leftover == null || leftover.Count > 1)
+			{
+				_logger.Error("Single AddItem return more than one stack");
+				return overflow;
+			}
+
+			// Append dictionary entry (1!) to overflow
+			if (leftover.Count > 0)
+			{
+				overflow.Add(index, leftover[0]);
+			}
+		}
+
+		return overflow;
+	}
+
+	public Dictionary<int, ItemStack> RemoveItem(params ItemStack[] items)
+	{
+		var remainders = new Dictionary<int, ItemStack>();
+		for (var index = 0; index < items.Length; index++)
+		{
+			var item = items[index];
+			var inventory = _inventories[item.Category];
+			var remainingStack = inventory.RemoveItem(item);
+
+			// Assertion, should never happen
+			if (remainingStack == null || remainingStack.Count > 1)
+			{
+				_logger.Error("Single AddItem return more than one stack");
+				return remainders;
+			}
+
+			// Append dictionary entry (1!) to overflow
+			if (remainingStack.Count > 0)
+			{
+				remainders.Add(index, remainingStack[0]);
+			}
+		}
+
+		return remainders;
+	}
+
+	public List<ItemStack> GetContents()
+	{
+		return _inventories.Values.SelectMany(inventory => inventory.GetContents()).ToList();
+	}
+
+	public void SetContents(List<ItemStack> items)
+	{
+		if (items.Count > Size)
+		{
+			_logger.Error("Too many items to fit in inventory");
+			return;
+		}
+
+		var index = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var sourceItems = items.GetRange(index, inventory.Size);
+			inventory.SetContents(sourceItems);
+			index += inventory.Size;
+		}
+	}
+
+	public bool Contains(string itemId)
+	{
+		return _inventories.Values.Any(inventory => inventory.Contains(itemId));
+	}
+
+	public bool Contains(ItemStack item)
+	{
+		return _inventories[item.Category].Contains(item);
+	}
+
+	public bool Contains(string itemId, int amount)
+	{
+		return _inventories.Values.Any(inventory => inventory.Contains(itemId, amount));
+	}
+
+	public bool Contains(ItemStack item, int amount)
+	{
+		return _inventories[item.Category].Contains(item, amount);
+	}
+
+	public bool ContainsAtLeast(ItemStack item, int amount)
+	{
+		return _inventories[item.Category].ContainsAtLeast(item, amount);
+	}
+
+	public Dictionary<int, ItemStack> All(string itemId)
+	{
+		var all = new Dictionary<int, ItemStack>();
+		var index = 0;
+		foreach (var inventory in _inventories.Values)
+		{
+			var items = inventory.All(itemId);
+			foreach (var (key, value) in items)
+			{
+				all.Add(key + index, value);
+			}
+
+			index += inventory.Size;
+		}
+
+		return all;
```

Comment:
Attempting...

Currently stuck on how to get the index offset.
Since the results from `simpleInventory.All(` are local indices, we need to convert them to "BigInventory indices"

Reply:
nevermind. its refering to `All` returning nothing